app/src/main/java/org/schabi/newpipe/streams/io/StoredFileHelper.java

```diff
@@ -285,7 +285,7 @@ public String getTag() {
     }
 
     public boolean existsAsFile() {
-        if (source == null) {
+        if (source == null || (docFile == null && ioFile == null)) {
             return false;
```

Comment:
maybe add a log line for debug builds that prints what was null? don't know if this might be helpful to analyse problems in the future